src/GeneratedMapper.Tests/ProjectToExtensionMethodTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿using GeneratedMapper.Tests.Helpers;
+using NUnit.Framework;
+
+namespace GeneratedMapper.Tests
+{
+    public class ProjectToExtensionMethodTests
+    {
+        [Test]
+        public void ProjectTo_With_NoMapAttributes_Only_Generates_ProjectTo_Single_Item()
+        {
+            GeneratorTestHelper.TestGeneratedCode(@"using System;
+using System.Linq;
+using System.Globalization;
+using System.Threading.Tasks;
+using GeneratedMapper.Attributes;
+
+[assembly: MapperGeneratorConfiguration(GenerateEnumerableMethods = true, GenerateExpressions = true)]
+namespace Test
+{
+    public class A { public int Id { get; set; }}
+    public class B { public int Id { get; set; }}
+
+    public class MapToClass { public B[] MapToCall() => new A[] { new() { Id = 5 } }.AsQueryable().ProjectTo<A,B>().ToArray(); }
+}",
+                @"using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Test.Expressions
+{
+    public static partial class A
+    {
+        public static Expression<Func<Test.A, Test.B>> ToB() => (Test.A self) =>
+            new Test.B
+            {
+                Id = self.Id,
+            };
+    }
+}
+",
+                GeneratorTestHelper.MapExtensionsDefaultText,
+                @"using System;
+using System.Linq;
+
+namespace GeneratedMapper.Extensions
+{
+    public static class MapExtension
+    {
+        public static IQueryable<TDestination> ProjectTo<TSource, TDestination>(this IQueryable<TSource> source)
+        {
+            switch (source)
+            {
+                case IQueryable<Test.A> a:
+                    return typeof(TDestination).FullName switch
+                    {
+                        ""Test.B"" =>
+                            a.Select(Test.Expressions.A.ToB()) is IQueryable<TDestination> b ? b : default,
+                        _ => throw new NotSupportedException(""Projection is not configured"")
```

Comment:
Can you add what projection is not configured, that could help with debugging.

Reply:
Added for both Projection and Map using TSource and TDestination.  They should never be able to hit this because compilation should fail before it gets this far, at least for now, but better safe then sorry.